pkg/datapath/iptables/iptables.go

```diff
@@ -1973,3 +2042,158 @@ func allEgressMasqueradeCmds(allocRange string, snatDstExclusionCIDR string,
 	}
 	return cmds
 }
+
+// hostNoTrackMultiPorts installs or removes a notrack rule matching multiple ports.
+// the use case for this is to skip conntrack when a pod uses hostNetwork to improve performance (pps/rps)
+// since conntrack affects the performance under load - which can occur under DDoS or traffic spikes for instance.
+func (m *Manager) hostNoTrackMultiPorts(prog iptablesInterface, cmd, proto string, ports []uint16) error {
+	// sort the slice containing the ports, and turn them into strings
+	slices.Sort(ports)
+	strPorts := make([]string, len(ports))
+	for i, p := range ports {
+		strPorts[i] = strconv.FormatUint(uint64(p), 10)
+	}
+
+	if err := prog.runProg([]string{
+		"-t", "raw",
+		cmd, ciliumPreRawChain,
+		"-p", strings.ToLower(proto),
+		"--match", "multiport",
+		"--dports", strings.Join(strPorts, ","),
+		"-m", "comment", "--comment", "cilium no-track-host-ports",
+		"-j", "CT",
+		"--notrack"}); err != nil {
+		return err
+	}
+
+	if err := prog.runProg([]string{
+		"-t", "raw",
+		cmd, ciliumOutputRawChain,
+		"-p", strings.ToLower(proto),
+		"--match", "multiport",
+		"--sports", strings.Join(strPorts, ","),
+		"-m", "comment", "--comment", "cilium no-track-host-ports return traffic",
+		"-j", "CT",
+		"--notrack"}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// groupL4AddrsByProto iterates over a slice of ports and returns a map with the port numbers
+// grouped by protocol.
+func groupL4AddrsByProto(ports []lb.L4Addr) map[lb.L4Type][]uint16 {
+	result := make(map[lb.L4Type][]uint16)
+
+	for _, p := range ports {
+		result[p.Protocol] = append(result[p.Protocol], p.Port)
+	}
+
+	return result
+}
+
+// replaceNoTrackHostPortRules replaces noTrackHostPort rules on a state change. the new ruleset is added, and the previous one is removed.
+func (m *Manager) replaceNoTrackHostPortRules(oldPorts, newPorts set.Set[lb.L4Addr]) error {
+	if oldPorts.Equal(newPorts) {
+		return nil
+	}
+
+	if err := m.installHostNoTrackRules(newPorts.AsSlice()); err != nil {
+		return err
+	}
+	if err := m.cleanupHostNoTrackRules(oldPorts.AsSlice()); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// installHostNoTrackRules installs a hostNoTrack multiport rule
+func (m *Manager) installHostNoTrackRules(p []lb.L4Addr) error {
+	byProto := groupL4AddrsByProto(p)
+
+	for proto, ports := range byProto {
+		if err := m.hostNoTrackMultiPorts(ip4tables, "-A", proto, ports); err != nil {
```

Comment:
Should this say: `if m.sharedCfg.EnableIPv4 {` given users could run Cilium also in IPv6-only mode.

Reply:
good point - will surely do that